meilisearch/index.py

```diff
@@ -43,60 +45,64 @@ def delete(self):
         return self.http.delete('{}/{}'.format(self.config.paths.index, self.uid))
 
     def update(self, **body):
-        """Update an index from meilisearch
+        """Update the index primary-key.
 
         Parameters
         ----------
         body: **kwargs
             Accepts primaryKey as an updatable parameter.
+            Ex: index.update(primaryKey='name')
 
         Returns
-        ----------
-        update: `dict`
-            Dictionnary containing an update id to track the action:
-            https://docs.meilisearch.com/references/updates.html#get-an-update-status
+        -------
+        index: dict
+            Dictionary containing index information.
         """
         payload = {}
         primary_key = body.get('primaryKey', None)
         if primary_key is not None:
             payload['primaryKey'] = primary_key
-        return self.http.put('{}/{}'.format(self.config.paths.index, self.uid), payload)
+        response = self.http.put('{}/{}'.format(self.config.paths.index, self.uid), payload)
+        self.primary_key = response['primaryKey']
+        return response
 
-    def info(self):
-        """Get info of index
+    def fetch_info(self):
+        """Fetch the info of the index.
 
         Returns
-        ----------
-        index: `dict`
-            Dictionnary containing index information.
+        -------
+        index : Index
+            An instance of Index containing the information of the index.
         """
-        return self.http.get('{}/{}'.format(self.config.paths.index, self.uid))
+        index_dict = self.http.get('{}/{}'.format(self.config.paths.index, self.uid))
+        self.primary_key = index_dict['primaryKey']
+        return self
 
     def get_primary_key(self):
-        """Get the primary key
+        """Get the primary key.
 
         Returns
-        ----------
+        -------
         primary_key: str
```

Comment:
```suggestion
        primary_key: str | None
```
What happens when the primary key was not yet defined? Does it return none ?

Reply:
Yes, it returns none.